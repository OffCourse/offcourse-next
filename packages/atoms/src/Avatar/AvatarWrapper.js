import system from "system-components";

const AvatarWrapper = system(
  {
    is: "div",
    bg: "black",
    hover: { backgroundColor: "primary" }
  },
  props => ({
    height: `${props.theme.avatars[props.variant].dimensions.height *
      props.multiply}rem`,
    width: `${props.theme.avatars[props.variant].dimensions.width *
      props.multiply}rem`,
    background: `url(${
      props.theme.avatars[props.variant].svg
    }) no-repeat top left`,
    backgroundColor: `${props.theme.avatars[props.variant].background}`,
    backgroundSize: "contain",
    boxSizing: "border-box"
  })
);

export default AvatarWrapper;
