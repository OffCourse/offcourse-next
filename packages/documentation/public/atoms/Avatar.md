Avatars can have a default or a custom image

```react|span-2
---
const name = "Yeehaa";
<Avatar name={name} />;
```

```react|span-2
const url = "https://assets.offcourse.io/portraits/offcourse_1.jpg";
const name = "Yeehaa";
<Avatar url={url} name={name} />;
```
