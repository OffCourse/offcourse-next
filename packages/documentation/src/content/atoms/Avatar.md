Avatar with Custom and Custom Image

```react|span-2
---
const name = "Yeehaa";
<Avatar name={name} />;
```

```react|span-2
const url = "https://assets.offcourse.io/portraits/offcourse_1.jpg";
const name = "Yeehaa";
<Avatar url={url} name={name} />;
```
