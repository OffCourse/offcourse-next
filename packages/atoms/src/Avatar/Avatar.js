import React, { Component } from "react";
import PropTypes from "prop-types";
import { lowerCase } from "../helpers";
import AvatarWrapper from "./AvatarWrapper";

/**
 * A Component for the Offcourse Project that shows the avatar image of a
 * given user
 */

class Avatar extends Component {
  static propTypes = {
    /**  url that refers to an avatar image */
    url: PropTypes.string,
    /** name of the person corresponding to this avatar */
    name: PropTypes.string.isRequired
  };

  static defaultProps = {
    url: "https://assets.offcourse.io/portraits/offcourse_2.jpg"
  };

  render() {
    const { url, name } = this.props;
    return <AvatarWrapper src={url} alt={`avatar of ${lowerCase(name)}`} />;
  }
}

export default Avatar;
