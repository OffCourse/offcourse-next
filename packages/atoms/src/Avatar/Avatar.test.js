import '../setupTests';

import React from 'react';
import ReactDOM from 'react-dom';
import system from 'system-components';
import { create as render } from 'react-test-renderer'
import { configure, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import { ThemeProvider } from "styled-components";

import Avatar from './Avatar';

describe('Avatar', () => {
  const props = {
    url: 'http://offcourse.io',
    name: 'Offcourse',
  };

  it('contains img', () => {
    const wrapper = mount(<Avatar {...props} />);

    expect(wrapper.find('img').length).toEqual(1);
  });

  it('has size 3em', () => {
    const wrapper = shallow(<Avatar {...props} />);

    expect(wrapper.props().size).toEqual('3em');
  });

  it('matches the snapshot', () => {
    const tree = shallow(<Avatar {...props} />);

    expect(toJson(tree)).toMatchSnapshot();
  });
});
