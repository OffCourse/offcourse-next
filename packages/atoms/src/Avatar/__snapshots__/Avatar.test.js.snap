// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar matches the snapshot 1`] = `
<Styled(div)
  alt="avatar of offcourse"
  is="img"
  size="3em"
  src="http://offcourse.io"
/>
`;
